*** streamproxy.c.orig	2017-03-13 12:19:43.425719954 +0000
--- streamproxy.c	2017-03-13 12:20:00.653725633 +0000
***************
*** 14,20 ****
  
  #define BSIZE                    1024*16
  
- #if DVB_API_VERSION < 5
  #define DMX_ADD_PID              _IO('o', 51)
  #define DMX_REMOVE_PID           _IO('o', 52)
  
--- 14,19 ----
***************
*** 22,28 ****
  	DMX_TAP_TS = 0,
  	DMX_TAP_PES = DMX_PES_OTHER, /* for backward binary compat. */
  } dmx_tap_type_t;
- #endif
  
  char response_line[MAX_LINE_LENGTH];
  int response_p;
--- 21,26 ----
