--- stblinux-2.6.12.orig/drivers/serial/8250.c	2009-01-20 18:17:12.000000000 +0100
+++ stblinux-2.6.12/drivers/serial/8250.c	2009-01-20 18:17:27.000000000 +0100
@@ -1800,6 +1800,9 @@
 	 */
 	baud = uart_get_baud_rate(port, termios, old, 0, port->uartclk/16); 
 	quot = serial8250_get_divisor(port, baud);
+	
+	quot = 5;
+	baud = 1000000;
 
 	/*
 	 * THT: Workaround for Ikos, forcing a value of quot = 14
@@ -1909,8 +1912,8 @@
 		serial_outp(up, UART_LCR, cval | UART_LCR_DLAB);/* set DLAB */
 	}
 
-	serial_outp(up, UART_DLL, quot & 0xff);		/* LS of divisor */
-	serial_outp(up, UART_DLM, quot >> 8);		/* MS of divisor */
+//	serial_outp(up, UART_DLL, quot & 0xff);		/* LS of divisor */
+//	serial_outp(up, UART_DLM, quot >> 8);		/* MS of divisor */
 
 	/*
 	 * LCR DLAB must be set to enable 64-byte FIFO mode. If the FCR
--- stblinux-2.6.12.orig/Makefile	2009-01-20 18:17:44.000000000 +0100
+++ stblinux-2.6.12/Makefile	2009-01-20 18:17:48.000000000 +0100
@@ -2,7 +2,7 @@
 PATCHLEVEL = 6
 SUBLEVEL = 12
 # STABLE_VERSION = .2
-EXTRAVERSION =-5.1-brcmstb-dm8000
+EXTRAVERSION =-5.1-brcmstb-dm500hd
 NAME=Woozy Numbat
 
 # *DOCUMENTATION*
--- stblinux-2.6.12.orig/arch/mips/brcmstb/lib/udivdi3.c	2009-01-20 18:23:12.000000000 +0100
+++ stblinux-2.6.12/arch/mips/brcmstb/lib/udivdi3.c	2009-01-20 18:23:16.000000000 +0100
@@ -32,7 +32,7 @@
 #include "gcclib.h"
 #include "longlong.h"
 
-static const UQItype __clz_tab[] =
+const UQItype __clz_tab[] =
 {
   0,1,2,2,3,3,3,3,4,4,4,4,4,4,4,4,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,
   6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,
--- stblinux-2.6.12-org/include/asm-mips/serial.h	2009-08-04 15:31:21.000000000 +0000
+++ stblinux-2.6.12/include/asm-mips/serial.h	2009-11-16 18:42:22.000000000 +0000
@@ -128,7 +128,7 @@
 #define BRCM_UART_PORT_DEFNS				\
 	_BRCM_16550_INIT(BRCM_SERIAL1_IRQ, BRCM_SERIAL1_BASE),		\
 	_BRCM_16550_INIT(BRCM_SERIAL2_IRQ, BRCM_SERIAL2_BASE),      \
-	_BRCM_16550_INIT(BRCM_SERIAL3_IRQ, BRCM_SERIAL3_BASE),	
+	_BRCM_16550_INIT(0, BRCM_SERIAL3_BASE),	
 
 #elif defined(CONFIG_MIPS_BCM7440) 
 #define BRCM_UART_PORT_DEFNS                            \
--- stblinux-2.6.12-org/drivers/mtd/brcmnand/brcmnand_base.c	2010-04-28 23:35:10.000000000 +0200
+++ stblinux-2.6.12/drivers/mtd/brcmnand/brcmnand_base.c	2010-04-29 00:07:48.000000000 +0200
@@ -332,10 +332,39 @@
 
 static unsigned char eccmask[64]; // Will be initialized during probe
 
+static uint32_t diff_p23[] = {
+	0x2848, 0x2850, 0x2854,			//2844..284c
+	0x2858, 0x2860, 0x2868, 0x286c,		//2850..285c
+	0x2870, 0x2874, 0x2878, 0x287c,		//2860..286c
+	0x2890, 0x2894, 0x2898, 0x289c,		//2870..287c
+	0x28a0, 0x28a4, 0x28a8, 0x28ac,		//2880..288c
+	0x28b0, 0x28b4, 0x28c0};		//2890..2898
+
+static int p23_init, p23_rev;
+
+static inline uint32_t __attribute__((always_inline)) fixreg_p23(uint32_t reg)
+{
+	if(unlikely(!p23_init)) {
+		int revid = *(volatile unsigned long *)0xb0404000;
+		p23_init = 1;
+		if( (revid >> 16) == 0x7413)
+			p23_rev = 1;
+	}
+
+	if(likely(!p23_rev || reg < 0x2844))
+		return reg;
+
+	return diff_p23[(reg-0x2844)>>2];
+}
+
 
 static uint32_t brcmnand_ctrl_read(uint32_t nandCtrlReg) 
 {
-	volatile unsigned long* pReg = (volatile unsigned long*) (BRCMNAND_CTRL_REGS 
+	volatile unsigned long* pReg;
+
+	nandCtrlReg = fixreg_p23(nandCtrlReg);
+
+	pReg = (volatile unsigned long*) (BRCMNAND_CTRL_REGS 
 		+ nandCtrlReg - BCHP_NAND_REVISION);
 
 	if (nandCtrlReg < BCHP_NAND_REVISION || nandCtrlReg > BCHP_NAND_BLK_WR_PROTECT ||
@@ -349,7 +378,11 @@
 
 static void brcmnand_ctrl_write(uint32_t nandCtrlReg, uint32_t val) 
 {
-	volatile unsigned long* pReg = (volatile unsigned long*) (BRCMNAND_CTRL_REGS 
+	volatile unsigned long* pReg;
+
+	nandCtrlReg = fixreg_p23(nandCtrlReg);
+
+	pReg = (volatile unsigned long*) (BRCMNAND_CTRL_REGS 
 		+ nandCtrlReg - BCHP_NAND_REVISION);
 
 	if (nandCtrlReg < BCHP_NAND_REVISION || nandCtrlReg > BCHP_NAND_BLK_WR_PROTECT ||
--- stblinux-2.6.12-org/drivers/net/brcmint7038/bcmemac.c	2010-04-28 23:35:03.000000000 +0200
+++ stblinux-2.6.12/drivers/net/brcmint7038/bcmemac.c	2010-04-29 00:07:47.000000000 +0200
@@ -3081,6 +3081,7 @@
 		case 0x7440:
 		case 0x7403:
 		case 0x7405:
+		case 0x7413:
 		break;
 	default:
 		printk(KERN_DEBUG CARDNAME" not found\n");
