diff -crB tuxtxt.orig/Makefile.am tuxtxt/Makefile.am
*** a/tuxtxt.orig/Makefile.am	2017-03-12 15:50:57.032044620 +0000
--- b/tuxtxt/Makefile.am		2017-03-12 16:11:49.632035808 +0000
***************
*** 32,37 ****
  	install -d $(DESTDIR)$(PLUGINDIR)
  	install -d $(DESTDIR)$(CONFIGDIR)
  	install -d $(DESTDIR)$(CONFIGDIR)/tuxtxt
! 	$(LIBTOOL) install tuxtxt.la $(DESTDIR)$(PLUGINDIR)
  	install -m 0644 $(srcdir)/tuxtxt.cfg $(DESTDIR)$(PLUGINDIR)
  	install -m 0644 $(srcdir)/tuxtxt.conf $(DESTDIR)$(CONFIGDIR)/tuxtxt
--- 32,37 ----
  	install -d $(DESTDIR)$(PLUGINDIR)
  	install -d $(DESTDIR)$(CONFIGDIR)
  	install -d $(DESTDIR)$(CONFIGDIR)/tuxtxt
! 	install tuxtxt.la $(DESTDIR)$(PLUGINDIR)
  	install -m 0644 $(srcdir)/tuxtxt.cfg $(DESTDIR)$(PLUGINDIR)
  	install -m 0644 $(srcdir)/tuxtxt.conf $(DESTDIR)$(CONFIGDIR)/tuxtxt
